packages/@react-spectrum/actiongroup/stories/ActionGroup.stories.tsx

```diff
@@ -156,3 +171,184 @@ function render(props = {}, items: any = itemsWithIcons) {
     </ActionGroup>
   );
 }
+let iconMap = {
+  'Document setup': DocumentIcon,
+  'Settings': SettingsIcon,
+  'Grid view': ViewGridIcon,
+  'List view': ViewListIcon,
+  'Gallery view': ViewCardIcon,
+  'Edit': DrawIcon,
+  'Copy': CopyIcon,
+  'Delete': DeleteIcon,
+  'Properties': PropertiesIcon,
+  'Info': InfoIcon,
+  'Keywords': BookIcon
+};
+function renderBoth(props, items: any = itemsWithIcons) {
+  return (
+    <ActionGroup onSelectionChange={s => onSelectionChange([...s])} {...props}>
+      {
+        items.map((itemProps) => {
+          let IconElement = iconMap[itemProps.children];
+          return (
+            <Item key={itemProps.name} textValue={itemProps.name}>
+              <Text>{itemProps.children}</Text>
+              <IconElement />
+            </Item>
+          );
+        })
+      }
+    </ActionGroup>
+  );
+}
+
+function renderIcons(props, items: any = itemsWithIcons) {
+  return (
+    <ActionGroup onSelectionChange={s => onSelectionChange([...s])} {...props}>
+      {
+        items.map((itemProps) => {
+          let IconElement = iconMap[itemProps.children];
+          return (
+            <Item key={itemProps.name} textValue={itemProps.name}>
+              <IconElement />
+            </Item>
+          );
+        })
+      }
+    </ActionGroup>
+  );
+}
+
+function renderBigOne() {
```

Comment:
🤯 
Although this is just a story, I'm think the `renderThing()` way of doing things is a React anti-pattern, and should probably be taken out into its own component.

Reply:
yeah, for whatever reason it's the storybook convention we've always had even back in v2
so unless you have an immediate concern, I'm probably not refactoring this in this pr

Reply:
yes, I realized that after.. definitely no concerns.. in fact, probably no biggie for stories in general